app/utils/pusher.js

```diff
@@ -63,29 +62,19 @@ TravisPusher.prototype.unsubscribeAll = function (channels) {
   return results;
 };
 
-TravisPusher.prototype.subscribe = function (channel) {
-  let ref;
-  if (!channel) {
-    return;
-  }
-  if (!((ref = this.pusher) != null ? ref.channel(channel) : void 0)) {
-    return this.pusher.subscribe(channel).bind_all((function (_this) {
-      return function (event, data) {
-        return _this.receive(event, data);
-      };
-    })(this));
+TravisPusher.prototype.subscribe = function (channelName) {
+  if (channelName && this.pusher && !this.pusher.channel(channelName)) {
+    this.active_channels.push(channelName);
```

Comment:
Is the order important here? i.e. does it make sense to add to the active channels only after the pusher subscription was successful?

Reply:
At the moment I just use it for tests, so it doesn't really matter I think. In production app we should rely on pusher as a source of truth. More explanation in commit message here: https://github.com/travis-ci/travis-web/pull/1275/commits/ecf87603f06057cb200222f45f1ee0754dd340d1

Reply:
Sorry, I read that initially and then when I started the review somehow lost track. Makes sense 👍